Assets/Scripts/States/TankPatrollingState.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class TankPatrollingState : TankState
+{
+    
+    private Vector3 tankStartingPosition;
+
+    private Vector3 spot2;
+    private Vector3 moveToSpot;
```

Comment:
Rename this as isTankAtStartingPosition
if(isTankAtStartingPosition == true) Move to Ending position
else Move to Starting position